jaxrs/src/test/java/io/airlift/jaxrs/TestRequestScopedBinding.java

```diff
@@ -0,0 +1,140 @@
+package io.airlift.jaxrs;
+
+import com.google.inject.BindingAnnotation;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.bootstrap.LifeCycleManager;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.Request;
+import io.airlift.http.server.HttpServerInfo;
+import io.airlift.http.server.testing.TestingHttpServerModule;
+import io.airlift.json.JsonModule;
+import io.airlift.node.testing.TestingNodeModule;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.HttpHeaders;
+import org.testng.annotations.Test;
+
+import java.lang.annotation.Retention;
+import java.net.URI;
+import java.util.function.Supplier;
+
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static io.airlift.jaxrs.JaxrsBinder.jaxrsBinder;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+public class TestRequestScopedBinding
+{
+    @Test
+    public void testRequestScopedInjection()
+    {
+        Injector injector = startServer();
+        try {
+            HttpClient httpClient = injector.getInstance(Key.get(HttpClient.class, ForTest.class));
+            URI serverUri = injector.getInstance(HttpServerInfo.class).getHttpUri();
+
+            assertEquals(executeRequest(httpClient, serverUri, "Bob"), "Bob@127.0.0.1");
+            assertEquals(executeRequest(httpClient, serverUri, "Alice"), "Alice@127.0.0.1");
+        }
+        finally {
+            injector.getInstance(LifeCycleManager.class).stop();
+        }
+    }
+
+    private String executeRequest(HttpClient httpClient, URI baseUri, String name)
+    {
+        Request request = prepareGet()
+                .setUri(baseUri.resolve("/hello"))
+                .setHeader("X-Name", name)
+                .build();
+        return httpClient.execute(request, createStringResponseHandler()).getBody();
+    }
+
+    @Path("/hello")
+    public static class MyResource
+    {
+        @GET
+        public String returnRequestScopedInfo(@Context RemoteAddr remoteAddr, @Context HeaderValue headerValue)
```

Comment:
This can be accomplished with standard JaxRs.

First you use `@BeanParam` instead `@Context`:
```java
public String returnRequestScopedInfo(@BeanParam RemoteAddr remoteAddr, @BeanParam HeaderValue headerValue)
```
Then for the classes you can do:
```java
    public record RemoteAddr(String value)
    {
        public RemoteAddr(@Context HttpServletRequest request)
        {
            this(request.getRemoteAddr());
        }
    }

    public static class HeaderValue
    {
        private final String value;

        private HeaderValue(@Context HttpHeaders headers)
        {
            this.value = headers.getHeaderString("X-Name");
        }

        public String getValue()
        {
            return value;
        }
    }
```